drgn/helpers/common/stack.py

```diff
@@ -0,0 +1,100 @@
+stack trace.
+"""
+
+from drgn import FaultError, PlatformFlags, StackTrace
+from drgn.helpers.common.memory import identify_address
+
+__all__ = ("print_annotated_stack",)
+
+
+def print_annotated_stack(trace: StackTrace) -> None:
+    """
+    Print the content of the stack trace, identifying symbols and
+    slab objects.
+
+    :param trace: Stack trace to print.
+    """
+    prog = trace.prog
+    # platform must be known if there is a stack trace
+    assert prog.platform is not None
+    is_64_bit = prog.platform.flags & PlatformFlags.IS_64_BIT
+    word_size = 8 if is_64_bit else 4
+    hex_format = "{:016x}" if is_64_bit else "{:08x}"
+
+    if prog.platform.flags & PlatformFlags.IS_LITTLE_ENDIAN:
+        byteorder = "little"
+    else:
+        byteorder = "big"
+
+    print("STACK POINTER     VALUE")
+
+    start = 0
+    while start < len(trace):
+        end = start + 1
+
+        while end < len(trace) and not trace[end].interrupted:
+            end += 1
+
+        # printing trace[start] through trace[end - 1]
+
+        frames = [trace[i] for i in range(start, end)]
+        frames_addrs = [frame.register("rsp") for frame in frames]
+
+        start_addr = frames_addrs[0]
+        end_addr = frames_addrs[-1] + word_size - 1
+        stack_size = end_addr - start_addr + 1
+
+        try:
+            stack_bytes = prog.read(start_addr, stack_size)
+        except FaultError:
+            # addresses not mapped
+            # for e.g, frames in userspace
+            for frame in frames:
+                print(f"[stack frame {frame}]")
+
+            start = end
+            continue
+
+        frame_ind = 0
+
+        for offset in range(0, len(stack_bytes), word_size):
+            addr = start_addr + offset
+            hex_addr = hex_format.format(addr)
+
+            word_bytes = stack_bytes[offset : offset + word_size]
+            word_val = int.from_bytes(
+                word_bytes,
+                # The byteorder parameter is annotated as Literal['little', 'big'], but
+                # mypy infers that our byteorder variable is str.
+                byteorder=byteorder,  # type: ignore[arg-type]
+            )
+
+            # multiple frames can have the same address
+            # for e.g, the first context_switch frame
+            while frame_ind < len(frames_addrs) and addr == frames_addrs[frame_ind]:
+                frame = frames[frame_ind]
+                print(f"[stack frame {frame}]")
+
+                frame_ind += 1
+            else:
+                formatted_value = hex_format.format(word_val)
+
+                identified = identify_address(prog, word_val)
+                if identified:
+                    formatted_value += f" [{identified}]"
+
+                if is_64_bit:
+                    print(f"{hex_addr}: {formatted_value}")
+                else:
+                    # 8 extra spaces to align the value
+                    print(f"{hex_addr}:         {formatted_value}")
```

Comment:
We can simplify this a lot if we define the format for the line based on 64-bit vs 32-bit at the beginning of the function:

```python3
    if prog.platform.flags & PlatformFlags.IS_64_BIT:
        line_format = "{:016x}: {:016x}{}"
    else:
        line_format = "{:08x}: {:08x}{}"
```

Then all you need here is:

```suggestion
                identified = identify_address(prog, word_val)
                if identified is None:
                    identified = ""
                else:
                    identified = f" [{identified}]"
                print(line_format.format(addr, word_val, identified))
```

(Then you can get rid of `hex_format` and `hex_addr` above, too.)